code/datums/components/storage/concrete/_concrete.dm

```diff
@@ -140,7 +140,13 @@
 	if(new_location)
 		//Reset the items values
 		_removal_reset(AM)
-		AM.forceMove(new_location)
+		if(!ismob(new_location || !isitem(AM)))
+			AM.forceMove(get_turf(new_location))
+		else
+			var/obj/item/target_item = AM
+			// not silent && mob should stand on a turf && item is on a proper turf. 'get_turf(loc)' will be sent to the proc to show the animation
+			// if fails any of those, pickup animation will be skipped
+			target_item.move_to_mobhand(new_location, !silent && isturf(new_location.loc) && get_turf(target_item.loc))
```

Comment:
This code is just bad

Reply:
How? This is the core of the issue we're having.

Reply:
Its weird flaky logic in the middle of the storage component. The first condition is also just wrong since the parenthesis are incorrect. The TG PR that you are porting from handled this in a much nicer way, so is there a reason why we can't just take that entirely?

Reply:
This weird flaky logic is mainly because of pickup animation. The animation doesn't know if you actually picked up or failed to pickup so it should show dropping animation.
seriously, I want to answer you better, but this has been months, and I am not able to tell you this accurately. all I can say is that the original code is already worse and causing a very very worse bug, and a refactor or this snowflaky fix will resolve that.

Well, back to the start... I've told you your `moved_relay` is broken, and that's because of this double picking up issue. the old code already does very very bad logic which causes a very annoying bug for years that made me and people fool.
The original entire logic in regard to pickup and moving items is incredibly bad, but also the current TG code uses /datum/storage, which is modularised version of the component storage system, but it is a big refactor that I don't want to deal with for now.
This means the whole logic is not changeable unless I attempt to port TG's new storage datum, but like I said, this isn't something I want to deal with, and porting will take ages. (I am willingly close this PR if that can happen)

Yeah, I know the answer is not clear... Literally the whole logic is already bad, so that I can't show you anything better for this - that's what I can say now.

Reply:
tldr; the item move logic already happens in two parts: the component and the mob's pickup proc - which is a god damn thing.